root/static/scripts/guess-case/modes.js

```diff
@@ -282,3 +283,27 @@ exports.Sentence = assign({}, DefaultMode, {
     {url: {href: 'https://musicbrainz.org/doc/Style/Language', target: '_blank'}}
   ),
 });
+
+exports.Turkish = assign({}, DefaultMode, {
+  description: l(
+    'This mode capitalises almost all words of a sentence, the words ' +
+    'which are restricted to be capitalised are lowercased. Some words, often proper nouns, ' +
+    'may need to be manually fixed according to the {url|Turkish ' +
+    'language guidelines}.',
+    {url: {href: 'https://musicbrainz.org/doc/Style/Language/Turkish', target: '_blank'}}
+  ),
+
+  isSentenceCaps() {
+    return false;
+  },
+
+  runPostProcess(is) {
+    return DefaultMode.runPostProcess(is)
+      .replace(/I/g, 'i') //replace 'I' with 'i' as 'I' is not valid in Turkish
+      //replace of Turkish lowercase words if they have been used at the end of sentence and capitalised
```

Comment:
`I` is a valid character in Turkish, but it is the upper case of `ı` (_dotless i_) whereas `İ` (_dotted I_) is the upper case of `i`. See https://musicbrainz.org/doc/Style/Language/Turkish